yast/lib/suse/connect/yast.rb

```diff
@@ -1,38 +1,39 @@
 require 'json'
-require 'ffi'
+require 'fiddle/import'
 require 'suse/toolkit/shim_utils'
 
 module Stdio
-  extend FFI::Library
-  ffi_lib FFI::Platform::LIBC
-  attach_function :free, [ :pointer ], :void
+  extend Fiddle::Importer
+  dlload 'libc.so.6'
+  typealias 'pointer', 'void*'
+  extern 'void free(pointer)'
 end
 
 module GoConnect
-  extend FFI::Library
-  ffi_lib 'suseconnect'
-
-  callback :log_line, [:int, :string], :void
-  attach_function :set_log_callback, [:log_line], :void
-
-  attach_function :announce_system, [:string, :string], :pointer
-  attach_function :update_system, [:string, :string], :pointer
-  attach_function :credentials, [:string], :pointer
-  attach_function :create_credentials_file, [:string, :string, :string], :pointer
-  attach_function :curlrc_credentials, [], :pointer
-  attach_function :show_product, [:string, :string], :pointer
-  attach_function :activated_products, [:string], :pointer
-  attach_function :activate_product, [:string, :string, :string], :pointer
-  attach_function :get_config, [:string], :pointer
-  attach_function :write_config, [:string], :pointer
-  attach_function :update_certificates, [], :pointer
-  attach_function :list_installer_updates, [:string, :string], :pointer
-  attach_function :system_migrations, [:string, :string], :pointer
-  attach_function :offline_system_migrations, [:string, :string, :string], :pointer
-  attach_function :upgrade_product, [:string, :string], :pointer
-  attach_function :synchronize, [:string, :string], :pointer
-  attach_function :system_activations, [:string], :pointer
-  attach_function :search_package, [:string, :string, :string], :pointer
+  extend Fiddle::Importer
+  dlload 'libsuseconnect.so'
+  typealias 'string', 'char*'
+
+  #callback type: void log_line(int, string)
```

Comment:
If it is not needed then I'd remove it completely, avoid dead code...

Reply:
This is not old code, just a plain comment. Fiddle doesn't support callback types so everything is passed as void pointer. Without this, future us would need to dig through the code to find out what is the expected function signature.

Reply:
Ah, I overlooked that, I thought it is part of the previous ffi code...